src/schemathesis/specs/openapi/negative/mutations.py

```diff
@@ -0,0 +1,153 @@
+    # one property
+    ordered_properties = draw(st.lists(st.sampled_from(properties), min_size=len(properties), unique_by=lambda x: x[0]))
+    for property_name, property_schema in ordered_properties:
+        if apply_mutations(draw, property_schema) == MutationResult.SUCCESS:
+            break
+    else:
+        # No successful mutations
+        return MutationResult.FAILURE
+    features = draw(st.shared(FeatureStrategy(), key="properties"))  # type: ignore
+    for name, property_schema in properties:
+        # Skip already mutated property
+        if name == property_name:  # pylint: disable=undefined-loop-variable
+            # Pylint: `properties` variable has at least one element as it is checked at the beginning of the function
+            # Then those properties are ordered and iterated over, therefore `property_name` is always defined
+            continue
+        # The `features` strategy is reused for property names and mutation names for simplicity as it is not likely
+        # to have an overlap between them.
+        if features.is_enabled(name):
+            for mutation in get_mutations(property_schema):
+                if features.is_enabled(mutation.__name__):
+                    mutation(draw, property_schema)
+    return MutationResult.SUCCESS
+
+
+def apply_mutations(draw: Draw, schema: Schema) -> MutationResult:
+    for mutation in get_mutations(schema):
+        if mutation(draw, schema) == MutationResult.SUCCESS:
+            return MutationResult.SUCCESS
+    return MutationResult.FAILURE
+
+
+def negate_constraints(draw: Draw, schema: Schema) -> MutationResult:
+    # TODO. implement constraints negation
+    return MutationResult.FAILURE
+
+
+def negate_schema(draw: Draw, schema: Schema) -> MutationResult:
+    """Negate the schema with JSON Schema's `not` keyword.
+
+    It is the least effective mutation as it negates the whole schema without trying to change its small parts.
+
+    Applicable types: any
+    """
+    if canonicalish(schema) == {}:
+        return MutationResult.FAILURE
+    inner = schema.copy()  # Shallow copy is OK
+    schema.clear()
+    schema["not"] = inner
+    return MutationResult.SUCCESS
+
+
+def get_mutations(schema: Schema) -> Tuple[Mutation, ...]:
+    """Get mutations possible for a schema.
+
+    Mutations are sorted by their anticipated effectiveness. The ordering is essential when we need to unconditionally
+    apply at least one mutation.
+    """
+    types = get_type(schema)
+    # On the top-level of Open API schemas, types are always strings, but inside "schema" objects, they are the same as
+    # in JSON Schema, where it could be either a string or an array of strings.
+    # TODO. How to handle multiple types? Maybe each mutation can have a guard that will return FAILURE on a type that
+    # is not applicable
+    if "object" in types:
+        return change_properties, negate_constraints, remove_required_property, change_schema_type, negate_schema
```

Comment:
On what basis this order is maintained?

Reply:
Mostly heuristic - from the most effective to the least effective. I still want to play & evaluate it, but the last mutation is negating everything, so it could be type, other constraints, etc., which seems like the least effective one, as it has the most broad spectrum of possible differences. I assume that smaller changes in the structure are better.

Reply:
> I assume that smaller changes in the structure are better.

I'd guess that "small" is the most effective single size, but that using a variety of sized changes finds even more bugs.

Reply:
> using a variety of sized changes finds even more bugs.

Could you, please, elaborate?

Reply:
Let's say that two endpoints allow invalid inputs of the correct type, and one allows invalid inputs only of the wrong type.

If we only make small changes (ie keep the type) we'll find two bugs, but if we sometimes keep the type and sometimes discard it we can find all three bugs.  ['Aim for maximum expressiveness in test cases'](https://blog.regehr.org/archives/591), even if we don't have full swarm-testing.

Reply:
Got it, thanks! 

I thought that this order is important only in the first step inside `change_properties` - when it is trying to apply at least one mutation to at least one property (to avoid cases when nothing is chosen by `FeatureStrategy` & it may be equivalent to the "positive" scenario). Other mutations to other properties are still possible in the second part of `change_properties`, depending on `FeatureStrategy`.

But now I see that in some cases, when there is only one property, then the latest mutation here might never be applied at all. E.g., for `{"foo": {"type": "string"}}`, it will always stop on `change_schema_type`. Probably randomized ordering might solve it, but maybe it could be weighted towards having `negate_schema` less probable?

Also, I completely forgot that applying `negate_schema` after others might cancel previous effects & lead to "positive" scenarios. E.g. `{"const": 42"}` will be negated twice, which gets us back to the same schema (however, it is the case with all schemas). So, probably some dependencies between mutations are needed.